vulnerabilities/pipelines/samba_importer.py

```diff
@@ -0,0 +1,335 @@
+            pre_tag = soup.find("pre")
+            if not pre_tag:
+                self.log(f"No detailed information found for {cve_id}", level=logging.WARNING)
+                return {}
+
+            announcement_text = html.unescape(pre_tag.get_text())
+            details = parse_announcement_text(announcement_text)
+            details["url"] = detail_url
+
+            self.advisory_details_cache[cve_id] = details
+            return details
+        except requests.exceptions.RequestException as e:
+            self.log(f"Error fetching advisory details for {cve_id}: {e}", level=logging.WARNING)
+            return {}
+        except Exception as e:
+            self.log(f"Error processing advisory details for {cve_id}: {e!r}", level=logging.ERROR)
+            return {}
+
+    def parse_advisory_row(self, row):
+        cells = row.find_all("td")
+        if len(cells) != 6:
+            self.log(
+                f"Expected 6 cells in security table row, got {len(cells)}", level=logging.WARNING
+            )
+            return None
+
+        date_issued = cells[0].get_text().strip()
+        patch_links = []
+
+        for link in cells[1].find_all("a"):
+            href = link.get("href", "")
+            if href:
+                if not href.startswith(("http://", "https://")):
+                    href = f"https://www.samba.org{href}"
+                patch_links.append(href)
+
+        issue_desc = cells[2].get_text().strip()
+        affected_releases = cells[3].get_text().strip()
+
+        cve_ids = []
+        for link in cells[4].find_all("a"):
+            cve_id = link.get_text().strip()
+            if re.match(r"CVE-\d{4}-\d{4,}", cve_id):
+                cve_ids.append(cve_id)
+
+        date_published = None
+        try:
+            if date_issued:
+                date_obj = dateparser.parse(date_issued)
+                if date_obj:
+                    date_published = date_obj.replace(tzinfo=pytz.UTC)
+        except Exception as e:
+            self.log(f"Error parsing date {date_issued}: {e!r}", level=logging.WARNING)
+
+        fixed_versions = self.extract_versions_from_patch_links(patch_links)
+        affected_packages = []
+
+        if fixed_versions:
+            base_purl = PackageURL(type="generic", name="samba")
+            for version in fixed_versions:
+                affected_packages.append(AffectedPackage(package=base_purl, fixed_version=version))
+        elif affected_releases and affected_releases != "Please refer to the advisories.":
+            base_purl = PackageURL(type="generic", name="samba")
+            version_match = re.search(r"(?:Samba|samba)\s+(\d+\.\d+\.\d+)", affected_releases)
+
+            if version_match:
+                affected_packages.append(
+                    AffectedPackage(
+                        package=base_purl, fixed_version=f"{version_match.group(1)}-fixed"
+                    )
+                )
```

Comment:
I honestly do not understand why would you put affected_version in fixed version.

And why are you mutating the original version by appending `-fixed` to version.